packages/unified-latex-util-pegjs/grammars/latex.pegjs

```diff
@@ -118,26 +118,66 @@ special_macro "special macro" // for the special macros like \[ \] and \begin{}
                 content: x.join(""),
             });
         }
+    // verbatim macro in listings package
+    / verbatim_listings
+    // verbatim macro in minted package
+    / verbatim_minted
     // verbatim environment
     / verbatim_environment
     // display math with \[...\]
     / begin_display_math
         x:(!end_display_math x:math_token { return x; })*
-        end_display_math { return createNode("displaymath", { content: x }); }
+        end_display_math { return createNode("displaymath", { content: x.flatMap(x => x) }); }
     // inline math with \(...\)
     / begin_inline_math
         x:(!end_inline_math x:math_token { return x; })*
-        end_inline_math { return createNode("inlinemath", { content: x }); }
+        end_inline_math { return createNode("inlinemath", { content: x.flatMap(x => x) }); }
     // display math with $$...$$
     / math_shift
         math_shift
         x:(!(math_shift math_shift) x:math_token { return x; })*
         math_shift
-        math_shift { return createNode("displaymath", { content: x }); }
+        math_shift { return createNode("displaymath", { content: x.flatMap(x => x) }); }
     // math with $...$
     / math_environment
     / environment
 
+square_bracket_argument
+    = "[" o:(!(end:. & { return end === "]"; }) x:token { return x; })* "]" {
```

Comment:
I think `end:.` should be `end:token` and the check will need to be `end.type === "string" && end.content === "]"`